087-external-certificate-manager.md

```diff
@@ -0,0 +1,241 @@
+
+In addition to Strimzi fully managing the certificates as described above, there are options for users to partially manage the certificates:
+* Users can [install and use their own CA certificate and private keys](https://strimzi.io/docs/operators/latest/deploying#installing-your-own-ca-certificates-str), instead of using the defaults generated by the Cluster Operator.
+  When using this option, both the CA certificate and private key must be provided, and Strimzi still issues the end-entity (EE) certificates that are presented by the components.
+* Users can [provide custom listener certificates](https://strimzi.io/docs/operators/latest/deploying#proc-installing-certs-per-listener-str) for TLS encryption.
+  This option only affects how user applications connect to Kafka.
+  It does not change how the Strimzi components connect to Kafka, or how the Kafka brokers connect to each other.
+
+None of the existing options allow all end-entity certificates to be issued by a tool that isn't Strimzi.
+
+## Motivation
+
+Strimzi's primary purpose is to provide a way to run Apache Kafka clusters on Kubernetes.
+Although it is nice that it can issue certificates, it would be beneficial if the certificates could be issued by a dedicated certificate manager, such as [cert-manager](https://cert-manager.io/).
+This is a feature that is often requested, especially because many organizations have specific compliance requirements with regard to certificates, for example:
+* Requiring that CA private keys are not shared.
+* Requiring that self-signed certificates cannot be used.
+
+## Proposal
+
+Strimzi will be updated to allow users to specify that certificates should be issued by an external certificate manager instead of issued by the Cluster Operator.
+This proposal will specifically describe how this would work for cert-manager, however the user API for configuration will be written in a way that does not prevent other external certificate managers being added in the future.
+
+The proposal makes a few assumptions:
+* Strimzi will not be responsible for installing cert-manager, but we will document the supported versions of cert-manager that we have tested with.
+* Strimzi will not be responsible for creating `Issuer` or `ClusterIssuer` custom resources.
+* Strimzi will not be responsible for requesting a CA certificate be issued, it will only interact with cert-manager to request end-entity certificates are issued.
+* Strimzi will create `Certificate` custom resources (to request a new end-entity certificate is issued) and will allow the user to influence the contents of these resources by exposing options in the `Kafka` custom resource.
+* Strimzi will not directly interact with the lower level `CertificateRequest` and `CertificateSigningRequests` custom resources.
+* When Strimzi creates a `Certificate` custom resource, cert-manager will issue the certificate within a reasonable amount of time such that Strimzi can wait during the reconciliation.
```

Comment:
I was also wondering, what happens if the cert-manager is down or is not working properly. The Strimzi operator cannot wait forever so the reconciliation would fail, right? Then because the `Certificate` was already created, the operator will take care of it in the next reconciliation still waiting for cert-manager until a timeout again?

Reply:
Can we clarify it with just one additional sentence?